src/one/nio/serial/GeneratedSerializer.java

```diff
@@ -38,13 +38,20 @@
     static final AtomicInteger migratedFields = new AtomicInteger();
     static final AtomicInteger renamedFields = new AtomicInteger();
     static final AtomicInteger unsupportedFields = new AtomicInteger();
+    static final Set<Class<?>> generationInProcess = new ConcurrentSkipListSet<Class<?>>(new Comparator<Class<?>>() {
+        @Override
+        public int compare(Class<?> o1, Class<?> o2) {
+            return o1.getName().compareTo(o2.getName());
+        }
+    });
```

Comment:
This would be incorrect with respect to Comparator contract. The relation must be symmetric and transitive. Anyway, Comarable-based collections won't work here. `ConcurrentHashMap<Class, Boolean>` is a simple and effective alternative to concurrent set. It can be explicitly transformed to a set using `Collections.newSetFromMap`, but this is not necessary here.